tango/integrations/beaker/workspace.py

```diff
@@ -331,25 +335,70 @@ def register_run(self, targets: Iterable[Step], name: Optional[str] = None) -> R
 
         return Run(name=cast(str, name), steps=steps, start_date=run_dataset.created)
 
-    def registered_runs(self) -> Dict[str, Run]:
-        import concurrent.futures
-
-        runs: Dict[str, Run] = {}
+    def search_registered_runs(
+        self,
+        *,
+        sort_by: WorkspaceSort = WorkspaceSort.START_DATE,
+        sort_descending: bool = True,
+        match: Optional[str] = None,
+        limit: Optional[int] = None,
+        cursor: Optional[int] = None,
```

Comment:
Cursors are nice, but they promise a lot. They promise that the underlying list didn't change between calls. Maybe we should't make that kind of promise, because it'll be hard to keep.

Reply:
The intention of this API is not to make that promise, which I think is fine. As a consequence, on the UI the data for a given page might change as you flip back and forth through pages, but I think that's okay.

Reply:
Yeah, while it would be super neat if we could support real cursors, it's not worth the effort.